src/components/SearchBar.tsx

```diff
@@ -31,61 +33,65 @@ type SearchBarCommandsType = {
   cancelSearch: (viewRef: React.ElementRef<typeof NativeSearchBar>) => void;
 };
 
-class SearchBar extends React.Component<SearchBarProps> {
-  nativeSearchBarRef: React.RefObject<SearchBarCommands>;
+const SearchBar = React.forwardRef<SearchBarCommands, SearchBarProps>(
+  function SearchBar(props, ref) {
+    const nativeSearchBarRef = React.useRef<SearchBarCommands | null>(null!);
+    React.useImperativeHandle(ref, () => ({
+      ...nativeSearchBarRef.current,
+      blur,
+      focus,
+      toggleCancelButton,
+      clearText,
+      setText,
+      cancelSearch,
+    }));
 
-  constructor(props: SearchBarProps) {
-    super(props);
-    this.nativeSearchBarRef = React.createRef();
-  }
-
-  _callMethodWithRef(method: (ref: SearchBarCommands) => void) {
-    const ref = this.nativeSearchBarRef.current;
-    if (ref) {
-      method(ref);
-    } else {
-      console.warn(
-        'Reference to native search bar component has not been updated yet'
-      );
+    function _callMethodWithRef(method: (ref: SearchBarCommands) => void) {
+      const ref = nativeSearchBarRef.current;
+      if (ref) {
+        method(ref);
+      } else {
+        console.warn(
+          'Reference to native search bar component has not been updated yet'
+        );
+      }
     }
-  }
 
-  blur() {
-    this._callMethodWithRef(ref => NativeSearchBarCommands.blur(ref));
-  }
+    function blur() {
+      _callMethodWithRef(ref => NativeSearchBarCommands.blur(ref));
+    }
 
-  focus() {
-    this._callMethodWithRef(ref => NativeSearchBarCommands.focus(ref));
-  }
+    function focus() {
+      _callMethodWithRef(ref => NativeSearchBarCommands.focus(ref));
+    }
 
-  toggleCancelButton(flag: boolean) {
-    this._callMethodWithRef(ref =>
-      NativeSearchBarCommands.toggleCancelButton(ref, flag)
-    );
-  }
+    function toggleCancelButton(flag: boolean) {
+      _callMethodWithRef(ref =>
+        NativeSearchBarCommands.toggleCancelButton(ref, flag)
+      );
+    }
 
-  clearText() {
-    this._callMethodWithRef(ref => NativeSearchBarCommands.clearText(ref));
-  }
+    function clearText() {
+      _callMethodWithRef(ref => NativeSearchBarCommands.clearText(ref));
+    }
 
-  setText(text: string) {
-    this._callMethodWithRef(ref => NativeSearchBarCommands.setText(ref, text));
-  }
+    function setText(text: string) {
+      _callMethodWithRef(ref => NativeSearchBarCommands.setText(ref, text));
+    }
 
-  cancelSearch() {
-    this._callMethodWithRef(ref => NativeSearchBarCommands.cancelSearch(ref));
-  }
+    function cancelSearch() {
+      _callMethodWithRef(ref => NativeSearchBarCommands.cancelSearch(ref));
+    }
```

Comment:
Won't these get recreated on every re-render? I think we should move the declaration to imperative handle or (if that makes sense in this use case) use callback.

Reply:
https://github.com/software-mansion/react-native-screens/pull/2132/commits/351a18702deb8873ad48cd9cec3686c29c26741c